ProcessSmarterTestPackage/PostProcessors/FormPartitionPostProcessor.cs

```diff
@@ -16,23 +16,101 @@ public override IList<ValidationError> GenerateErrors()
 
             var partitionIdentifier = Processor.ChildNodeWithName("identifier").ValueForAttribute("uniqueid");
             var itemGroups = Processor.ChildNodesWithName("itemgroup").ToList();
+            const string groupIdPattern = @"(\d+-\d+):([G|I])-(\d+-\d+)";
+
             foreach (var itemGroup in itemGroups)
             {
-                if (!itemGroup.ChildNodeWithName("identifier")
-                    .ValueForAttribute("uniqueid").Split(':').First()
-                    .Equals(partitionIdentifier, StringComparison.OrdinalIgnoreCase))
+                var match = Regex.Match(itemGroup.ChildNodeWithName("identifier")
```

Comment:
The changes here emit errors around group identifiers. In the Student application, groups identifiers whose second portion begins with "G-" are treated differently than those that start with "I-". Other identifier prefixes are not recognized. By convention, the first part of a group ID (the part preceding the colon) should be the form partition ID that the group belongs to. The part following should be either the associated stimuli ID found in the passageref element (for "G-" item groups) or the item ID of the single item in the group (for "I-" groups). Errors emitted as "Severe" will cause the application to crash at runtime when it attempts to load those groups. Benign errors here are violations of the convention for group IDs that have no material effect on loading and administering assessments.